sdk/monitor/azure-monitor-query/azure/monitor/query/_metrics_batch_query_client.py

```diff
@@ -0,0 +1,143 @@
+            tuple[~datetime.datetime, ~datetime.datetime]]]
+        :keyword granularity: The granularity (i.e. timegrain) of the query.
+        :paramtype granularity: Optional[~datetime.timedelta]
+        :keyword aggregations: The list of aggregation types to retrieve. Use
+            `azure.monitor.query.MetricAggregationType` enum to get each aggregation type.
+        :paramtype aggregations: Optional[list[str]]
+        :keyword max_results: The maximum number of records to retrieve.
+            Valid only if $filter is specified. Defaults to 10.
+        :paramtype max_results: Optional[int]
+        :keyword order_by: The aggregation to use for sorting results and the direction of the sort.
+            Only one order can be specified. Examples: sum asc.
+        :paramtype order_by: Optional[str]
+        :keyword filter: The **$filter** is used to reduce the set of metric data returned. Example:
+            Metric contains metadata A, B and C. - Return all time series of C where A = a1 and B = b1 or
+            b2 **$filter=A eq 'a1' and B eq 'b1' or B eq 'b2' and C eq '*'** - Invalid variant: **$filter=A
+            eq 'a1' and B eq 'b1' and C eq '*' or B = 'b2'** This is invalid because the logical or
+            operator cannot separate two different metadata names. - Return all time series where A = a1,
+            B = b1 and C = c1: **$filter=A eq 'a1' and B eq 'b1' and C eq 'c1'** - Return all time series
+            where A = a1 **$filter=A eq 'a1' and B eq '*' and C eq '*'**. Special case: When dimension
+            name or dimension value uses round brackets. Eg: When dimension name is **dim (test) 1**
+            Instead of using **$filter= "dim (test) 1 eq '*'"** use **$filter= "dim %2528test%2529 1 eq '*'"**.
+            When dimension name is **dim (test) 3** and dimension value is **dim3 (test) val**, instead of using
+            **$filter= "dim (test) 3 eq 'dim3 (test) val'"** use **$filter= "dim
+            %2528test%2529 3 eq 'dim3 %2528test%2529 val'"**. Default value is None.
+        :paramtype filter: str
+        :return: A list ofMetricsQueryResult objects.
+        :rtype: list[~azure.monitor.query.MetricsQueryResult]
+        :raises ~azure.core.exceptions.HttpResponseError
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_metrics_batch_query.py
+                :start-after: [START send_metrics_batch_query]
+                :end-before: [END send_metrics_batch_query]
+                :language: python
+                :dedent: 0
+                :caption: Get a response for a batch metrics query.
+        """
+        if not resource_uris:
+            raise ValueError("resource_uris must be provided and must not be empty.")
+        if not metric_namespace:
+            raise ValueError("metric_namespace must be provided and must not be empty.")
+        if not metric_names:
```

Comment:
is this level of validation necessary to construct the uri? Just wondering, because we don't usually do this level of client side validation unless we'd get a misleading error otherwise